### PR TITLE
[New Nav] Fix: Automatically apply Colony details updates to UI

### LINKED ISSUE #3096: [New Navigation] Refetch user's joined colonies information after colony info updates
**Description**

When the colony details are updated via an action, the Colony Switcher isn't refetching the latest data for a user's joined colonies list. 

**Completion checklist**

* [ ] The colony logo should be updated in the Colony Switcher when a Colony's avatar is updated Image
* [ ] The colony display name should be correctly shown and should update after a successful Colony Details Edit action 
image

### PR DESCRIPTION
**Pull request by rumzledz**
## Description

Update avatar

## Testing

1. Open the Colony Switcher and take note of the your Colony's avatar
2. Bring up the Action Form and set it to Edit Colony Details
3. Upload a different Colony avatar
4. Edit the Colony name
5. Fill in all other required form fields and submit the form
6. After the Action is successfully completed, open the Colony Switcher
7. Verify that the Colony Avatar is updated
8. Verify that the new Colony name is used for the Colony Switcher

Resolves #3096

### CODE REVIEW

File: src/components/v5/shared/Navigation/ColonySwitcher/ColonySwitcher.tsx [hunk 1]
```diff
@@ -47,9 +47,12 @@ const ColonySwitcher: React.FC<ColonySwitcherProps> = ({
...
           />
-          {colonyContext?.colony.name && (
+          {colonyContext?.colony && (
             <p className="truncate text-md font-semibold text-base-white">
-              {capitalizeFirstLetter(colonyContext.colony.name)}
+              {capitalizeFirstLetter(
+                colonyContext.colony.metadata?.displayName ??
+                  colonyContext.colony.name,
+              )}
             </p>
```
Comment (mmioana) [inline src/components/v5/shared/Navigation/ColonySwitcher/ColonySwitcher.tsx:56]: I would propose to create a constant 
`const colonyName = colonyContext?.colony.metadata?.displayName ?? colonyContext?.colony.name`
and then refactor these lines as
```
{colonyName && (
   <p className="truncate text-md font-semibold text-base-white">
       {capitalizeFirstLetter(colonyName)}
   </p>
 )}
```

File: src/components/v5/shared/Navigation/ColonySwitcher/partials/JoinedColonyItem/JoinedColonyItem.tsx [hunk 2]
```diff
@@ -44,7 +44,7 @@ const JoinedColonyItem = ({
       </div>
       <div className="flex w-full flex-col overflow-hidden text-left">
         <p className="truncate text-md font-semibold text-gray-900">
-          {capitalizeFirstLetter(name)}
+          {capitalizeFirstLetter(metadata?.displayName ?? name)}
```
Comment (mmioana) [inline src/components/v5/shared/Navigation/ColonySwitcher/partials/JoinedColonyItem/JoinedColonyItem.tsx:47]: Also here let's make use of a constant

**Review by mmioana — APPROVED** [review]
You're on fire fixing all these issues @rumzledz 🔥 

Nice work, tested and can confirm updating the colony name and avatar have effect also for the colony switcher and sidebar menu header section

Before
Screenshot 2024-09-17 at 12 17 41
Changing the logo
Screenshot 2024-09-17 at 12 18 42
Screenshot 2024-09-17 at 12 18 48
Editing the colony name
Screenshot 2024-09-17 at 12 19 48
Screenshot 2024-09-17 at 12 20 01

Left a couple of refactoring comments, but nothing blocking this PR to get merged 💪

File: src/components/v5/shared/Navigation/ColonySwitcher/ColonySwitcher.tsx (hunk 1 above)
Comment (rumzledz) [reply src/components/v5/shared/Navigation/ColonySwitcher/ColonySwitcher.tsx:56]: All right lemme do that 👌

**Review by rdig — APPROVED** [review]
Nicely done. Works as expected! :100: 

Screenshot from 2024-09-18 12-23-10
Screenshot from 2024-09-18 12-24-14
Screenshot from 2024-09-18 12-24-22
Screenshot from 2024-09-18 12-24-28

**Review by bassgeta — APPROVED** [review]
Nice and simple, though this really makes me miss react query's query invalidation mechanism, it's so much cleaner than invoking refetch :'(
Before submitting the action:
image
As soon as it changes, it's refetched!
image

Nice one :ok_hand:

**Pull request merged**